pkg/ingester/limiter_test.go

```diff
@@ -295,13 +335,97 @@ func runLimiterMaxFunctionTest(
 			overrides, err := validation.NewOverrides(limits, nil)
 			require.NoError(t, err)
 
-			limiter := NewLimiter(overrides, ring, testData.ringReplicationFactor, testData.ringZoneAwarenessEnabled, "zone")
+			strategy := newIngesterRingLimiterStrategy(ring, testData.ringReplicationFactor, testData.ringZoneAwarenessEnabled, "zone", overrides.IngestionTenantShardSize)
+			limiter := NewLimiter(overrides, strategy)
+			actual := runMaxFn(limiter)
+			assert.Equal(t, testData.expectedValue, actual)
+		})
+	}
+}
+
+func runLimiterMaxFunctionTestWithPartitionsRing(
+	t *testing.T,
+	applyLimits func(limits *validation.Limits, globalLimit int),
+	runMaxFn func(limiter *Limiter) int,
+	expectedLimitForZeroGlobalLimit int,
+) {
+	tests := map[string]struct {
+		globalLimit               int
+		partitionStates           []ring.PartitionState
+		tenantPartitionsShardSize int
+		expectedValue             int
+	}{
+		"limit is disabled": {
+			globalLimit:               0,
+			partitionStates:           []ring.PartitionState{ring.PartitionActive, ring.PartitionInactive},
+			tenantPartitionsShardSize: 0,
+			expectedValue:             expectedLimitForZeroGlobalLimit,
+		},
+		"limit is enabled, using all active partitions, only 1 is active": {
+			globalLimit:               1000,
+			partitionStates:           []ring.PartitionState{ring.PartitionActive, ring.PartitionInactive, ring.PartitionInactive, ring.PartitionInactive, ring.PartitionInactive},
+			tenantPartitionsShardSize: 0,
+			expectedValue:             1000, // 1000 / 1
+		},
+		"limit is enabled, using all active partitions, only 2 are active": {
+			globalLimit:               1000,
+			partitionStates:           []ring.PartitionState{ring.PartitionActive, ring.PartitionActive, ring.PartitionInactive, ring.PartitionInactive, ring.PartitionInactive},
+			tenantPartitionsShardSize: 0,
+			expectedValue:             500, // 1000 / 2
```

Comment:
[nit] I like showing the intention, but maybe you can just replace the expected value with the comment content.

```suggestion
			expectedValue:             1000 / 2,
```

_Same comment applies in other cases of this test._

Reply:
Done in https://github.com/grafana/mimir/pull/7424/commits/9811e539683e5ab52d353d1e3d0533992561ed60.